src/Controller/ModulesController.php

```diff
@@ -477,6 +481,39 @@ public function relationshipsJson($id, string $relation) : void
         $this->set('_serialize', array_keys($response));
     }
 
+    /**
+     * Retrieve thumbnails URL of related objects in `meta.url` if present.
+     *
+     * @param array $response Related objects response.
+     * @return void
+     */
+    public function getThumbsUrls(array &$response) : void
+    {
+        if (empty($response['data'])) {
+            return;
+        }
+
+        // extract ids of objects
+        $ids = Hash::extract($response, 'data.{n}.id');
```

Comment:
[superminor] add array cast `(array)Hash::extract(...)`

Reply:
no mention in cake3 doc, what is the reason to be that specific?

Reply:
scrutinizer suggestion, but we can skip it...

Reply:
i was just curious